pkg/github/teamreadwriter.go

```diff
@@ -20,29 +20,49 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/google/go-github/v61/github"
 	"google.golang.org/protobuf/proto"
 
+	"github.com/abcxyz/pkg/cache"
 	"github.com/abcxyz/pkg/sets"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
-const IDSep = ":"
+const (
+	IDSep = ":"
+	// DefaultCacheDuration is the default time to live for the user and team caches.
+	// We don't expect user info (e.g. username etc.) nor team info (team name etc.)
+	// to change frequently so a time to live of 1 day is the default.
+	DefaultCacheDuration = time.Hour * 24
+)
 
 type OrgTokenSource interface {
 	// TokenForOrg returns a token that grants access to the given Org's resources.
 	TokenForOrg(ctx context.Context, orgID int64) (string, error)
 }
 
-type Opt func(writer *TeamReadWriter)
+type Config struct {
+	includeSubTeams bool
+	cacheDuration   time.Duration
+}
+
+type Opt func(writer *Config)
 
 // WithoutSubTeamsAsMembers toggles off treating subteams as members of their parent team.
 // When this option is used TeamReadWriter.GetMembers will only return user members of the team.
 // Similarly, TeamReadWriter.SetMembers will only consider user members when setting members.
 func WithoutSubTeamsAsMembers() Opt {
-	return func(writer *TeamReadWriter) {
-		writer.includeSubTeams = false
+	return func(config *Config) {
+		config.includeSubTeams = false
+	}
+}
+
+// WithCacheDuration set the time to live for the user and team cache entries.
+func WithCacheDuration(duration time.Duration) Opt {
```

Comment:
Is there a place where we are feeding this in as a runtime config? I'd like it to be an environment variable that we can tweak at runtime.

Reply:
Not yet, we are currently importing this library to an internal service. We will be standing up a version of TeamLink in GCP but that has yet to be written. I think what you are suggesting makes a lot of sense but is probably better to have what ever code instantiates the `TeamReadWriter` to worry about sourcing the cache duration from an environment variable.